peers.go

```diff
@@ -15,68 +15,87 @@ limitations under the License.
 */
 
 // peers.go defines how processes find and communicate with their peers.
+// Each running Universe instance is a peer of each other, and it has authority over a set of keys, or "stars", within each galaxy (address space of data) -- which keys are handled by each peer is determined by the consistent hashing algorithm. Each instance fetches from another peer when it receives a request for a key for which that peer is the authority.
 
-package groupcache
+package galaxycache
 
 import (
 	"context"
+	"sync"
 
-	pb "github.com/golang/groupcache/groupcachepb"
+	"github.com/vimeo/groupcache/consistenthash"
+	pb "github.com/vimeo/groupcache/groupcachepb"
 )
 
-// ProtoGetter is the interface that must be implemented by a peer.
-type ProtoGetter interface {
-	Get(context context.Context, in *pb.GetRequest, out *pb.GetResponse) error
+// RemoteFetcher is the interface that must be implemented to fetch from other peers; the PeerPicker contains a map of these fetchers corresponding to each other peer address
+type RemoteFetcher interface {
+	Fetch(context context.Context, in *pb.GetRequest, out *pb.GetResponse) error
 }
 
-// PeerPicker is the interface that must be implemented to locate
-// the peer that owns a specific key.
-type PeerPicker interface {
-	// PickPeer returns the peer that owns the specific key
-	// and true to indicate that a remote peer was nominated.
-	// It returns nil, false if the key owner is the current peer.
-	PickPeer(key string) (peer ProtoGetter, ok bool)
+// PeerPicker is in charge of dealing with peers: it contains the hashing options (hash function and number of replicas), consistent hash map of peers, and a map of RemoteFetchers to those peers
+type PeerPicker struct {
+	fetchingProtocol FetchProtocol
+	selfURL          string
+	peers            *consistenthash.Map
+	fetchers         map[string]RemoteFetcher
+	mu               sync.RWMutex
+	opts             HashOptions
 }
 
-// NoPeers is an implementation of PeerPicker that never finds a peer.
-type NoPeers struct{}
+// HashOptions specifies the the hash function and the number of replicas for consistent hashing
+type HashOptions struct {
+	// Replicas specifies the number of key replicas on the consistent hash.
+	// If blank, it defaults to 50.
```

Comment:
```suggestion
	// If zero, it defaults to 50.
```
(in go, blank gives a ["zero-value"])


["zero-value"]: https://golang.org/ref/spec#The_zero_value